cmd/operator/api/v1alpha1/kubearchiveconfig_webhook.go

```diff
@@ -13,65 +14,86 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 
-	cel "github.com/kubearchive/kubearchive/pkg/cel"
+	"github.com/kubearchive/kubearchive/pkg/cel"
 )
 
-const kubearchiveResourceName = "kubearchive"
-
 // log is for logging in this package.
 var kubearchiveconfiglog = logf.Log.WithName("kubearchiveconfig-resource")
 
-// SetupWebhookWithManager will setup the manager to manage the webhooks
-func (kac *KubeArchiveConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(kac).
+		For(&KubeArchiveConfig{}).
+		WithValidator(&KubeArchiveConfigCustomValidator{kubearchiveResourceName: "kubearchive"}).
+		WithDefaulter(&KubeArchiveConfigCustomDefaulter{}).
 		Complete()
 }
 
 //+kubebuilder:webhook:path=/mutate-kubearchive-kubearchive-org-v1alpha1-kubearchiveconfig,mutating=true,failurePolicy=fail,sideEffects=None,groups=kubearchive.kubearchive.org,resources=kubearchiveconfigs,verbs=create;update,versions=v1alpha1,name=mkubearchiveconfig.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Defaulter = &KubeArchiveConfig{}
+type KubeArchiveConfigCustomDefaulter struct{}
+
+var _ webhook.CustomDefaulter = &KubeArchiveConfigCustomDefaulter{}
 
-// Default implements webhook.Defaulter so a webhook will be registered for the type
-func (kac *KubeArchiveConfig) Default() {
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (kaccd *KubeArchiveConfigCustomDefaulter) Default(_ context.Context, obj runtime.Object) error {
+	kac, ok := obj.(*KubeArchiveConfig)
+	if !ok {
+		return fmt.Errorf("expected an KubeArchiveConfig object but got %T", obj)
+	}
 	kubearchiveconfiglog.Info("default", "namespace", kac.Namespace, "name", kac.Name)
+	return nil
 }
 
 //+kubebuilder:webhook:path=/validate-kubearchive-kubearchive-org-v1alpha1-kubearchiveconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=kubearchive.kubearchive.org,resources=kubearchiveconfigs,verbs=create;update,versions=v1alpha1,name=vkubearchiveconfig.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Validator = &KubeArchiveConfig{}
+type KubeArchiveConfigCustomValidator struct {
+	kubearchiveResourceName string
+}
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (kac *KubeArchiveConfig) ValidateCreate() (admission.Warnings, error) {
-	kubearchiveconfiglog.Info("validate create", "namespace", kac.Namespace, "name", kac.Name)
+var _ webhook.CustomValidator = &KubeArchiveConfigCustomValidator{}
 
-	if kac.Name != kubearchiveResourceName {
-		return nil, fmt.Errorf("The KubeArchiveConfig resource must be named '%s'.", kubearchiveResourceName)
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (kaccv *KubeArchiveConfigCustomValidator) ValidateCreate(
+	_ context.Context, obj runtime.Object) (admission.Warnings, error) {
```

Comment:
This is a nit, but I think this would be easier to read if the function declaration can't fit on one line

```suggestion
func (kaccv *KubeArchiveConfigCustomValidator) ValidateCreate(
	_ context.Context, obj runtime.Object,
) (admission.Warnings, error) {
```

Reply:
agree! fixed